public/app/features/alerting/unified/rule-list/GroupedView.test.tsx

```diff
@@ -0,0 +1,44 @@
+import { http } from 'msw';
+import { render, screen } from 'test/test-utils';
+
+import { setPluginComponentsHook, setPluginLinksHook, setReturnToPreviousHook } from '@grafana/runtime';
+import { AccessControlAction } from 'app/types';
+
+import { setupMswServer } from '../mockApi';
+import { grantUserPermissions } from '../mocks';
+import { alertingFactory } from '../mocks/server/db';
+import { paginatedHandlerFor } from '../mocks/server/utils';
+
+import { GroupedView } from './GroupedView';
+
+setPluginLinksHook(() => ({ links: [], isLoading: false }));
+setPluginComponentsHook(() => ({ components: [], isLoading: false }));
+setReturnToPreviousHook(() => () => {});
+
+grantUserPermissions([AccessControlAction.AlertingRuleExternalRead]);
+
+const server = setupMswServer();
+
+const mimirGroups = alertingFactory.group.buildList(2000, { file: 'test-mimir-namespace' });
+alertingFactory.group.rewindSequence();
+const prometheusGroups = alertingFactory.group.buildList(200, { file: 'test-prometheus-namespace' });
+
+const mimirDs = alertingFactory.dataSource.build({ name: 'Mimir', uid: 'mimir' });
+const prometheusDs = alertingFactory.dataSource.build({ name: 'Prometheus', uid: 'prometheus' });
+
+beforeEach(() => {
+  server.use(http.get(`/api/prometheus/${mimirDs.uid}/api/v1/rules`, paginatedHandlerFor(mimirGroups)));
+  server.use(http.get(`/api/prometheus/${prometheusDs.uid}/api/v1/rules`, paginatedHandlerFor(prometheusGroups)));
```

Comment:
Can we please move these elsewhere so the files don't reference the `server` specifically? If needed, it could go into a helper method that sets up the server in a certain predefined way, or ideally every test gets the same logic (as the APIs should be behaving this way anyway)

Something like the other methods in `mocks/server/configure.ts`?